src/python/pants/util/objects.py

```diff
@@ -46,12 +47,8 @@ def make_type_error(cls, msg, *args, **kwargs):
     """
     return cls.type_check_error_type(cls.__name__, msg, *args, **kwargs)
 
-  @abstractmethod
-  def copy(self, **kwargs):
-    """Return a new object of the same type, replacing specified fields with new values"""
-
 
-# TODO(#7074): Migrate to python 3 dataclasses!
+@deprecated('1.24.0.dev2', hint_message='Use @dataclass to declare typed named tuples instead!')
```

Comment:
We'll want to remove this without a deprecation cycle once possible under the same logic as the `enum()` removal, i.e. that this API was experimental (albeit highly useful!).

Maybe hold off on deprecating this so that we can have a dedicated PR for the removal.

Reply:
I would think that deprecating it is exactly what makes it easier to remove eventually? If we don't want to have a full deprecation cycle, we can kill it early. But if we're planning to remove it, I think a deprecation cycle sounds reasonable?